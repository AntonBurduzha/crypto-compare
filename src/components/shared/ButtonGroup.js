import React from 'react';
import { Button } from 'antd';

export const ButtonGroup = ({ startConnection, stopConnection, isConnected }) => {
  return (
    <div className="cc-socket-wrapper">
      <Button
        type="primary"
        size="large"
        style={{ width: '120px', height: '40px', marginRight: '20px' }}
        onClick={startConnection}
        disabled={isConnected}
      >
        Start
      </Button>
      <Button
        type="danger"
        size="large"
        style={{ width: '120px', height: '40px' }}
        onClick={stopConnection}
        disabled={!isConnected}
      >
        Stop
      </Button>
    </div>
  );
}
