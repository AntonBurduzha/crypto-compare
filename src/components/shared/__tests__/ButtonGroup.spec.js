import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button } from 'antd';
import { ButtonGroup } from '../ButtonGroup';

describe('ButtonGroup component', () => {
  it('renders without crashing', () => {
    const wrap = shallow(<ButtonGroup/>);
    expect(wrap).toBeDefined();
  });

  it('has 2 buttons', () => {
    const wrap = mount(<ButtonGroup/>);
    expect(wrap.find(Button)).toHaveLength(2);
    expect(wrap.find(Button).at(0).text()).toEqual('Start');
    expect(wrap.find(Button).at(1).text()).toEqual('Stop');
  });

  it('matches snapshot', () => {
    const wrap = shallow(<ButtonGroup/>);
    expect(wrap).toMatchSnapshot();
  })
});
