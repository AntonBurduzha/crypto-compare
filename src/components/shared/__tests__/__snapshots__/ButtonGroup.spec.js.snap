// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          disabled={undefined}
          ghost={false}
          loading={false}
          onClick={undefined}
          prefixCls="ant-btn"
          size="large"
          style={
                    Object {
                              "height": "40px",
                              "marginRight": "20px",
                              "width": "120px",
                            }
          }
          type="primary"
>
          Start
</Button>,
        <Button
          disabled={true}
          ghost={false}
          loading={false}
          onClick={undefined}
          prefixCls="ant-btn"
          size="large"
          style={
                    Object {
                              "height": "40px",
                              "width": "120px",
                            }
          }
          type="danger"
>
          Stop
</Button>,
      ],
      "className": "cc-socket-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Start",
          "disabled": undefined,
          "ghost": false,
          "loading": false,
          "onClick": undefined,
          "prefixCls": "ant-btn",
          "size": "large",
          "style": Object {
            "height": "40px",
            "marginRight": "20px",
            "width": "120px",
          },
          "type": "primary",
        },
        "ref": null,
        "rendered": "Start",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Stop",
          "disabled": true,
          "ghost": false,
          "loading": false,
          "onClick": undefined,
          "prefixCls": "ant-btn",
          "size": "large",
          "style": Object {
            "height": "40px",
            "width": "120px",
          },
          "type": "danger",
        },
        "ref": null,
        "rendered": "Stop",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            disabled={undefined}
            ghost={false}
            loading={false}
            onClick={undefined}
            prefixCls="ant-btn"
            size="large"
            style={
                        Object {
                                    "height": "40px",
                                    "marginRight": "20px",
                                    "width": "120px",
                                  }
            }
            type="primary"
>
            Start
</Button>,
          <Button
            disabled={true}
            ghost={false}
            loading={false}
            onClick={undefined}
            prefixCls="ant-btn"
            size="large"
            style={
                        Object {
                                    "height": "40px",
                                    "width": "120px",
                                  }
            }
            type="danger"
>
            Stop
</Button>,
        ],
        "className": "cc-socket-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Start",
            "disabled": undefined,
            "ghost": false,
            "loading": false,
            "onClick": undefined,
            "prefixCls": "ant-btn",
            "size": "large",
            "style": Object {
              "height": "40px",
              "marginRight": "20px",
              "width": "120px",
            },
            "type": "primary",
          },
          "ref": null,
          "rendered": "Start",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Stop",
            "disabled": true,
            "ghost": false,
            "loading": false,
            "onClick": undefined,
            "prefixCls": "ant-btn",
            "size": "large",
            "style": Object {
              "height": "40px",
              "width": "120px",
            },
            "type": "danger",
          },
          "ref": null,
          "rendered": "Stop",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
